torch/csrc/autograd/init.cpp

```diff
@@ -62,7 +62,7 @@ PyObject* THPAutograd_initExtension(PyObject* _unused, PyObject *unused) {
   m.def(
       "_call_end_callbacks_on_fut",
       [](const at::Tensor& handle,
-         const std::shared_ptr<torch::distributed::rpc::FutureIValue>& fut) {
+         const c10::intrusive_ptr<c10::ivalue::Future>& fut) {
```

Comment:
Profiler function changes look good; this change is essentially making it so that `_call_end_callbacks_on_fut` gets called via pybind from python eager mode, and as a JIT operator in Torchscript

Reply:
I think you can also remove the `rpc/message.h` header in this file now?

Reply:
Doing that.